File: frontend/src/Types/sprint.ts
import type { Guid } from './api';

export interface Sprint {
  id: Guid;
  name: string;
  startDate: string;
  endDate: string;
  teamId: Guid;
  meetingIds: Guid[];
}

export interface CreateSprintRequest {
  name: string;
  startDate: string;
  endDate: string;
  teamId: Guid;
}

export interface UpdateSprintRequest {
  name: string;
  startDate: string;
  endDate: string;
  teamId: Guid;
}

File: backend/src/ScrumLog.Api/Data/Entities/Sprint.cs
namespace ScrumLog.Api.Data.Entities;

public class Sprint
{
    public Guid Id { get; set; }

    public required string Name { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public Guid TeamId { get; set; }

    public Team Team { get; set; } = null!;

    public ICollection<Meeting> Meetings { get; set; } = new List<Meeting>();
}
